brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/ToscaFacade.java

```diff
@@ -0,0 +1,41 @@
+package io.cloudsoft.tosca.a4c.brooklyn;
+
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.apache.brooklyn.config.ConfigKey;
+
+import com.google.common.base.Optional;
+
+import alien4cloud.component.repository.exception.CSARVersionNotFoundException;
+
+public interface ToscaFacade<A extends  ToscaApplication> {
+
+    Optional<Path> getArtifactPath(String nodeId, A toscaApplication, String artifactId);
+
+    Optional<Iterable<String>> getArtifacts(String nodeId, A toscaApplication);
+
+    Path getCsarPath(String archiveName, String archiveVersion) throws CSARVersionNotFoundException;
+
+    ConfigKey<String> getLifeCycle(String opKey);
+
+    String getParentId(String nodeId, A toscaApplication);
+
+    Optional<Map<String, Object>> getPropertiesAndTypeValues(String nodeId, A toscaApplication, String requirementId, String computeName);
```

Comment:
Is there a semantic difference between `Optional.absent()` and `Collections.emptyMap()`?